hello_world/CMakeLists.txt

```diff
@@ -1,46 +1,55 @@
-# The main CMake version check happens in the root CMakeLists.txt; if that
-# passed the current version is fine so just use it
-cmake_minimum_required(VERSION "${CMAKE_VERSION}")
-
-project(HelloWorld LANGUAGES CXX)
-
-# Used to deduce the directories for installation
-include(GNUInstallDirs)
-
-# Library names should be filesystem friendly name (so not case-sensitive)
-set(lib_name hello_world)
-
-# Declares a one-source library
-add_library(${lib_name} "${CMAKE_CURRENT_SOURCE_DIR}/src/hello-world.cpp")
-
-# Declares that are library has one public header file
-set_target_properties(
-    ${lib_name} PROPERTIES PUBLIC_HEADER
-    "${CMAKE_CURRENT_SOURCE_DIR}/include/hello-world.hpp"
-)
-
-# Tells CMake where header files the target depends on can be found
-target_include_directories(
-    ${lib_name} PUBLIC
-    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include/>
-    $<INSTALL_INTERFACE:include/>
-)
-
-# Installs the library and header files and associates it with an export target
-# "${lib_name}-targets"
-install(
-    TARGETS ${lib_name}
-    EXPORT "${lib_name}-targets"
-    PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_FULL_INCLUDEDIR}
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-    LIBRARY DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}
-    RUNTIME DESTINATION ${CMAKE_INSTALL_FULL_BINDIR}
-)
-
-# Installs the export target
-install(
-    EXPORT ${lib_name}-targets
-    NAMESPACE ${lib_name}::
-    FILE ${lib_name}Config.cmake
-    DESTINATION ${CMAKE_INSTALL_FULL_LIBDIR}/cmake
-)
+# The main CMake version check happens in the root CMakeLists.txt; if that
+# passed the current version is fine so just use it
+cmake_minimum_required(VERSION "${CMAKE_VERSION}")
+
+project(HelloWorld LANGUAGES CXX)
+
+# Can be generalized to CMAKE_PREFIX_PATH for multiple dependencies
+set(CMAKE_PREFIX_PATH "${CMAKE_INSTALL_PREFIX}/lib/cmake")

```

Comment:
This should be passed in via the `ExternalProject_Add` command. Also you very rarely want to "set" a variable which starts with CMAKE. Those are the variables end users will use to control the build and if you set them you overwrite whatever value the end user set it to.